generator/types.go

```diff
@@ -934,6 +934,7 @@ func (t *typeResolver) ResolveSchema(schema *spec.Schema, isAnonymous, isRequire
 			break
 		}
 		result.HasDiscriminator = schema.Discriminator != ""
+		result.Extensions = schema.Extensions
```

Comment:
This is probably tampering with a lot of other extensions. Have you considered adding the x-go-tag extension in the setExtension func (called in a defer statement, so it is not overwritten by ref resolution)?

Reply:
Based on your original idea (which is good) I have a tried a version that pass the unit tests and seems to do the job, at least on a basic test case.

```go
+++ b/generator/types.go
@@ -1244,6 +1244,13 @@ func (rt *resolvedType) setExtensions(schema *spec.Schema, origType string) {
        rt.IsEnumCI = hasEnumCI(schema.Extensions)
        rt.setIsEmptyOmitted(schema, origType)
        rt.setIsJSONString(schema, origType)
+
+       if customTag, found := schema.Extensions[xGoCustomTag]; found {
+               if rt.Extensions == nil {
+                       rt.Extensions = make(spec.Extensions)
+               }
+               rt.Extensions[xGoCustomTag] = customTag
+       }
 }
 
 func (rt *resolvedType) setIsEmptyOmitted(schema *spec.Schema, tpe string) {
```

```go
+++ b/generator/model.go
@@ -917,7 +917,7 @@ func (sg *schemaGenContext) buildProperties() error {
                emprop.GenSchema.Extensions = emprop.Schema.Extensions
 
                // set custom serializer tag
-               if customTag, found := emprop.Schema.Extensions[xGoCustomTag]; found {
+               if customTag, found := tpe.Extensions[xGoCustomTag]; found {
                        emprop.GenSchema.CustomTag = customTag.(string)
                }
                sg.GenSchema.Properties = append(sg.GenSchema.Properties, emprop.GenSchema)
@@ -1649,7 +1649,7 @@ func (sg *schemaGenContext) buildAdditionalItems() error {
 }
 
```

Reply:
We have to be more surgical regarding bubbling up extensions, because of the x-nullable.